### PR TITLE
Fix NDArray.mgrid and add unit tests. Also corrects the Object compar…

### PR DESCRIPTION
**Pull request by JetstreamRoySprowl**
Fix NDArray.mgrid and add unit tests. Also corrects the Object comparer for NDArray to compare shape and data contents.

### CODE REVIEW

**Comment by JetstreamRoySprowl** [comment]
Resolves https://github.com/SciSharp/NumSharp/issues/313

File: test/NumSharp.UnitTest/README.md [hunk 1]
```diff
@@ -7,4 +7,52 @@ Install `Codecov` from https://www.nuget.org/packages/Codecov.
...
+`PS D:\Projects\NumSharp> codecov -f coverage.xml -t <codecov-numsharp-token>`
+
+
+### NDArray generation for Unit Test cases
+
+Generating NDArray constants by hand is tedious and error prone, so here
+is some Python code that spits out C# definitions for static NDArrays
+that can be used in unit tests:
+
+```python
```
Comment (Nucs) [inline test/NumSharp.UnitTest/README.md]: The generating code should be placed in `NumSharp.UnitTest/Utilities/ndarray-generator.py` (or named anything of that sort)

File: test/NumSharp.UnitTest/README.md [hunk 2]
```diff
@@ -7,4 +7,52 @@ Install `Codecov` from https://www.nuget.org/packages/Codecov.
...
+                if commasToPrint > 0:
+                    rowStr = rowStr + ", "
+                commasToPrint -= 1
+                #if (i < v)
+            print(rowStr)
+        print("            }}, new Shape(new int[] {{ {}, {} }}));".format(v.shape[0], v.shape[1]))
+        print("")
+
+# generate mgrid unit test data (we're trusting Python here)
+aa, bb = np.mgrid[0:5, 0:3]
```
Comment (Nucs) [inline test/NumSharp.UnitTest/README.md]: The code that was used to generate you can place as a comment in `Creation/NdArray.Mgrid.Test.cs`

**Comment by JetstreamRoySprowl** [comment]
Updated and squashed with all requested changes.

**Comment by JetstreamRoySprowl** [comment]
Thanks, @Nucs ! Is there a chance of getting the NuGet pushed up right away? My project is using Azure IoT Edge devices which get built in docker containers, so there's no good way to get this update (which is on my critical path) without nuget.

**Comment by Nucs** [comment]
@Oceania2018, can you release a minor update for NumSharp?

**Comment by Oceania2018** [comment]
Sure, Before releasing, I have to fix another issue. I need to sign the library.
`Referenced assembly 'NumSharp.Core, Version=0.10.3.0, Culture=neutral, PublicKeyToken=null' does not have a strong name. `

**Comment by Nucs** [comment]
@JetstreamRoySprowl `0.10.4.0` was published.

**Comment by JetstreamRoySprowl** [comment]
@Nucs , @Oceania2018 , just what I needed -- thanks so much!

**Pull request merged**